exercise/static/exercise/chapter.js

```diff
@@ -168,56 +179,168 @@
+			button.attr("class", "btn btn-primary");
+			button.attr("value", "Submit");
+			button.attr("type", "submit");
+
+			$(first_div).append(label, form_field);
+			$(second_div).append(button);
+			$(form).append(first_div, second_div);
+			$(wrap).append(form);
+
+			return $(wrap);
 		},
 
 		load: function() {
 			this.showLoader("load");
 			var exercise = this;
-			$.ajax(this.url, {dataType: "html"})
-				.fail(function() {
-					exercise.showLoader("error");
-					exercise.chapter.nextExercise();
-				})
-				.done(function(data) {
-					exercise.hideLoader();
-					exercise.update($(data));
-					if (exercise.quiz) {
-						exercise.loadLastSubmission($(data));
-					} else {
+
+			if (exercise.settings.input) {
+				var title = exercise.element.data("title");
+				var type = exercise.element.data("type");
+				var def_val = exercise.element.data("default");
+
+				if (!title) title = '';
+				if (!def_val) def_val = '';
+
+				exercise.hideLoader();
+				var input_form = exercise.makeInputForm(exercise.chapterID, title, type, def_val);
+				exercise.update(input_form);
+				exercise.loadLastSubmission(input_form);
+				exercise.chapter.nextExercise();
+			 } else {
+				$.ajax(this.url, {dataType: "html"})
+					.fail(function() {
+						exercise.showLoader("error");
 						exercise.chapter.nextExercise();
-					}
-				});
+					})
+					.done(function(data) {
+						exercise.hideLoader();
+						exercise.update($(data));
+						if (exercise.quiz || exercise.active_element) {
+							exercise.loadLastSubmission($(data));
+						} else {
+							exercise.chapter.nextExercise();
+						}
+					 });
+				}
 		},
 
 		update: function(input) {
-			var content = this.element.find(this.settings.content_selector)
-				.empty().append(
-					input.filter(this.settings.exercise_selector).contents()
-				);
+			var exercise = this;
+			input = input.filter(exercise.settings.exercise_selector).contents();
+			var content = exercise.element.find(exercise.settings.content_selector)
+				.empty().append(input).hide();
+
+			if (exercise.active_element) {
+				var title = "";
+				if (exercise.element.attr("data-title"))
+					title = "<p><b>" + exercise.element.attr("data-title") + "</b></p>";
+				exercise.element.find(exercise.settings.summary_selector).remove();
+				$(title).prependTo(exercise.element.find(".exercise-response"));
+			}
+
+			content.show();
+
+			// Active element can have height settings in the A+ exercise div that need to be
+			// attached to correct DOM-elements before setting the exercise container div height to auto
+			var cur_height = exercise.element.css('height');
+			if (exercise.active_element) {
+				if (exercise.settings.input) {
+					$("#" + exercise.chapterID + " textarea").css("height", cur_height);
+				} else {
+					if (typeof $("#" + exercise.chapterID).data("scale") != "undefined") {
+						var cont_height = $(exercise.settings.ae_result_selector, exercise.element)[0].scrollHeight;
```

Comment:
I think I had a case in which the line 330 with `.scrollHeight` crashed due to an undefined object when the connection to the exercise service failed. Did you check what happens when the response from the exercise service does not contain the normally expected HTML structure due to an error?